examples/AN00221_freq_domain_example/src/AN00218_freq_domain_example.xc

```diff
@@ -4,32 +4,106 @@
 #include <stdlib.h>
 #include <stdint.h>
 #include <string.h>
+#include <stdio.h>
+#include <math.h>
 
 #include "mic_array.h"
 
 #include "lib_dsp.h"
 
-#define DF 2    //Decimation Factor
+#include "i2c.h"
+#include "i2s.h"
+
+#include "xscope.h"
+
+#define DSP_ENABLED 1
+#define INPUT_FROM_HW 1
+// MIC_ARRAY_WORD_LENGTH_SHORT 0 defined in mic_array_conf.h
+
+
+#define DECIMATION_FACTOR   2   //Corresponds to a 48kHz output sample rate
+#define DECIMATOR_COUNT     2   //8 channels requires 2 decimators
+#define NUM_FRAME_BUFFERS   3   //Triple buffer needed for overlapping frames
 
 on tile[0]: in port p_pdm_clk               = XS1_PORT_1E;
 on tile[0]: in buffered port:32 p_pdm_mics  = XS1_PORT_8B;
 on tile[0]: in port p_mclk                  = XS1_PORT_1F;
 on tile[0]: clock pdmclk                    = XS1_CLKBLK_1;
 
+out buffered port:32 p_i2s_dout[1]  = on tile[1]: {XS1_PORT_1P};
+in port p_mclk_in1                  = on tile[1]: XS1_PORT_1O;
+out buffered port:32 p_bclk         = on tile[1]: XS1_PORT_1M;
+out buffered port:32 p_lrclk        = on tile[1]: XS1_PORT_1N;
+port p_i2c                          = on tile[1]: XS1_PORT_4E; // Bit 0: SCLK, Bit 1: SDA
+port p_rst_shared                   = on tile[1]: XS1_PORT_4F; // Bit 0: DAC_RST_N, Bit 1: ETH_RST_N
+clock mclk                          = on tile[1]: XS1_CLKBLK_3;
+clock bclk                          = on tile[1]: XS1_CLKBLK_4;
+
+
+
 #define FFT_N (1<<MIC_ARRAY_MAX_FRAME_SIZE_LOG2)
 
-int data[8][THIRD_STAGE_COEFS_PER_STAGE*DF];
+int data[8][THIRD_STAGE_COEFS_PER_STAGE*DECIMATION_FACTOR];
+
+#define NUM_SIGNAL_ARRAYS MIC_ARRAY_NUM_MICS/2
+
+#define NUM_OUTPUT_CHANNELS 2
+
+typedef struct {
+    int32_t data[NUM_OUTPUT_CHANNELS][FFT_N/2]; // FFT_N/2 due to overlapping
+} multichannel_audio_block_s;
+
+
+/**
+ The interface between the two tasks is a single transaction that get_next_bufs
+ the movable pointers. It has an argument that is a reference to a
+ movable pointer. Since it is a reference the server side of the
+ connection can update the argument.
+*/
+
+interface bufget_i {
+  void get_next_buf(multichannel_audio_block_s * unsafe &buf);
+};
+
+// make global to enforce 64 bit alignment
+multichannel_audio_block_s triple_buffer[3];
+
 
 int your_favourite_window_function(unsigned i, unsigned window_length){
-    return INT_MAX; //Boxcar
+    //Hanning function takes 10k memory which blows up the memory with FFT_N 4k
+    //TODO: Optimise using lib_dsp_math_sqrt and lib_dsp_math_cos.
```

Comment:
This TODO is about saving memory, not improving performance.
See previous comment

> +    //Hanning function takes 10k memory which blows up the memory with FFT_N 4k

By using the lib_dsp functions the memory will be reduced.

Is it not OK to have TODOs in the code during development phase?
Where else should they go?

Thanks, /T

From: Andrew Stanford-Jason <notifications@github.com<mailto:notifications@github.com>>
Reply-To: xmos/lib_mic_array <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Wednesday, 11 May 2016 15:25
To: xmos/lib_mic_array <lib_mic_array@noreply.github.com<mailto:lib_mic_array@noreply.github.com>>
Cc: thomas <thomas@xmos.com<mailto:thomas@xmos.com>>, Author <author@noreply.github.com<mailto:author@noreply.github.com>>
Subject: Re: [xmos/lib_mic_array] Improved AN00221_freq_domain_example (#27)


In examples/AN00221_freq_domain_example/src/AN00218_freq_domain_example.xc<https://github.com/xmos/lib_mic_array/pull/27#discussion_r62846640>:

>
>  int your_favourite_window_function(unsigned i, unsigned window_length){
> -    return INT_MAX; //Boxcar
> +    //Hanning function takes 10k memory which blows up the memory with FFT_N 4k
> +    //TODO: Optimise using lib_dsp_math_sqrt and lib_dsp_math_cos.


There is no need for this TODO. It is a one time execution. Also TODO don't go into app notes

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/xmos/lib_mic_array/pull/27/files/128603ba1525eed620fb83a8976e2964dc24d3ff#r62846640>